nautobot/extras/management/__init__.py

```diff
@@ -389,3 +395,82 @@ def _clear_custom_role_or_status_instances(
 
     if verbosity >= 2:
         print(f"    Deleted {deleted_total}, unlinked {unlinked_total} {metadatamodel} records")
+
+
+def validate_job_and_job_data(command, user, job_class_path, data=None):
+    job_data = {}
+    try:
+        if data:
+            job_data = json.loads(data)
+    except json.decoder.JSONDecodeError as error:
+        raise CommandError(f"Invalid JSON data:\n{error!s}") from error
+
+    if not get_job(job_class_path, reload=True):
+        raise CommandError(f'Job "{job_class_path}" not found')
+
+    try:
+        job_model = Job.objects.get_for_class_path(job_class_path)
+    except Job.DoesNotExist as error:
+        raise CommandError(f"Job {job_class_path} does not exist.") from error
+
+    try:
+        job_model = Job.objects.restrict(user, "run").get_for_class_path(job_class_path)
+    except Job.DoesNotExist:
+        raise CommandError(f"User {user.username} does not have permission to run this Job") from None
+
+    if not job_model.installed or job_model.job_class is None:
+        raise CommandError("Job is not presently installed")
```

Comment:
This *might* be partially redundant with lines 408-409?

Reply:
I am a little hesitant to modify it since it was in the original `runjob.py`. Since these lines are checking if the job is installed and lines 408-409 are checking whether the job can be imported?

Reply:
Fine to leave as-is.